tools/gulp/tasks/examples.ts

```diff
@@ -0,0 +1,247 @@
+import {ExampleMaterialModule} from './material-module';
+
+export interface LiveExample {
+  title: string;
+  component: any;
+  additionalFiles?: string[];
+  selectorName?: string;
+}
+
+${extractedMetadata.map(r => buildImportsTemplate(r)).join('').trim()}
+
+export const EXAMPLE_COMPONENTS = {
+  ${extractedMetadata.map(r => buildExamplesTemplate(r)).join('').trim()}
+};
+
+export const EXAMPLE_LIST = [
+  ${extractedMetadata.map(r => buildListTemplate(r)).join('').trim()}
+];
+
+@NgModule({
+  declarations: EXAMPLE_LIST,
+  entryComponents: EXAMPLE_LIST,
+  imports: [
+    ExampleMaterialModule,
+    FormsModule,
+    ReactiveFormsModule,
+    CommonModule
+  ]
+})
+export class ExampleModule { }
+`;
+}
+
+/**
+ * Given a string that is a camel or pascal case,
+ * this function will convert to dash case.
+ * 
+ * @param {string} name 
+ * @returns {string} 
+ */
+function convertToDashCase(name: string): string {
+  name = name.replace(/[A-Z]/g, ' $&');
+  name = name.toLowerCase().trim();
+  return name.split(' ').join('-')
+}
+
+/**
+ * Parse the AST of a file and get metadata about it
+ * 
+ * @param {string} filename 
+ * @param {string} src 
+ * @returns {{ primary: any, seconds: any[] }} 
+ */
+function createMetas(filename: string, src: string): { primaryComponent: any, secondaryComponents: any[] } {
+  const sourceFile = ts.createSourceFile(
+    filename, src, ts.ScriptTarget.Latest, false, ts.ScriptKind.TS);
+
+  const metas: any[] = [];
+
+  const visit = (node: any): void => {
+    if (node.kind === ts.SyntaxKind.ClassDeclaration) {
+      const meta: any = {
+        component: node.name.text
+      };
+
+      let primary = false;
+      if(node.jsDoc && node.jsDoc.length) {
+        for(const doc of node.jsDoc) {
+          if(doc.tags && doc.tags.length) {
+            for(const tag of doc.tags) {
+              const tagValue = tag.comment;
+              const tagName = tag.tagName.text;
+              if(tagName === 'title') {
+                meta.title = tagValue;
+                meta.primary = true;
+              }
+            }
+          }
+        }
+      }
+
+      if (node.decorators && node.decorators.length) {
+        for (const decorator of node.decorators) {
+          if (decorator.expression.expression.text === 'Component') {
+            for (const arg of decorator.expression.arguments) {
+              for (const prop of arg.properties) {
+                const name = prop.name.text;
+                const value = prop.initializer.text;
+                meta[name] = value;
+              }
+            }
+          }
+        }
+      }
+
+      metas.push(meta);
+    }
+
+    ts.forEachChild(node, visit);
+  };
+
+  visit(sourceFile);
+
+  return {
+    primaryComponent: metas.find(m => m.primary),
+    secondaryComponents: metas.filter(m => !m.primary)
+  };
+}
+
+/**
+ * Creates the examples module and metadata
+ */
+task('build-examples-module', (cb) => {
```

Comment:
Ya, you need to wait until all operations are completed before we mark the task completed.